Products/ZenHub/PBDaemon.py

```diff
@@ -1210,6 +1217,12 @@ def buildOptions(self):
                                type='int',
                                help='Maximum number of events to queue')
 
+        self.parser.add_option('--maxqueueage',
```

Comment:
In general I try to keep changes to anything other than Products/ZenPackAdapter out of this branch.    I'd suggest getting this into develop (and merge it in here too), just to make it easier to keep things reconciled.

Reply:
I can isolate it - I'm just not sure whether or not they'd want an age-limit added to the event queue. I can ask Jason about this too.

Without these changes, two-thirds of the buffer has to be filled before the daemon will send.  (The default buffer size is 5000.)  Event if we set a low (via configs,) a few events could wait around a long time before the daemon sends them.

Reply:
This particular change looks reasonably isolated from the rest of the ZPA project to call it an enhancement for pushing events based on age (in addition to the queue size).

Reply:
Interesting, that 5 second loop makes sense though.
I wonder if that particular code isn't running in ZPA, because when I working on events, they never were sent until I added this

Reply:
The `pushEventsLoop` code should have been scheduled into the Twisted reactor when PBDaemon's `run` method is called.

Reply:
This is all correct now.  I updated monkey.py to correctly use and backed these changes to pbdaemon. 
`@monkeypatch(PBDaemon)
def pushEventsLoop(self):
    pass`